Serializer/PlainTextDeserializationVisitor.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Smartbox\CoreBundle\Serializer;
+
+use JMS\Serializer\AbstractVisitor;
+use JMS\Serializer\Context;
+use JMS\Serializer\Exception\RuntimeException;
+use JMS\Serializer\GraphNavigator;
+use JMS\Serializer\Metadata\ClassMetadata;
+use JMS\Serializer\Metadata\PropertyMetadata;
+
+/**
+ * This is a really simple visitor that will effectively bypass JMS. Useful for REST APIs that are not truly REST APIs
+ * and return plain text responses, which JMS by default refuses to process because they are not real JSON.
+ */
+class PlainTextDeserializationVisitor extends AbstractVisitor
+{
+    public function visitNull($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Null types.');
+    }
+
+    public function visitString($data, array $type, Context $context)
+    {
+        return $data;
+    }
+
+    public function visitBoolean($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Boolean types.');
+    }
+
+    public function visitDouble($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Double types.');
+    }
+
+    public function visitInteger($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Integer types.');
+    }
+
+    public function visitArray($data, array $type, Context $context)
+    {
+        throw new RuntimeException('PlainTextDeserializationVisitor cannot visit Array types.');
+    }
+
+    public function startVisitingObject(ClassMetadata $metadata, $data, array $type, Context $context)
+    {
+        // noop
+    }
+
+    public function visitProperty(PropertyMetadata $metadata, $data, Context $context)
+    {
+        // noop
+    }
+
+    public function endVisitingObject(ClassMetadata $metadata, $data, array $type, Context $context)
+    {
+        // noop
+    }
+
+    public function setNavigator(GraphNavigator $navigator)
+    {
+        // noop
+    }
+
+    public function getNavigator()
+    {
+        // noop
+    }
+
+    public function getResult()
+    {
+        return null;
```

Comment:
Some particular reason of defining this as return null, rather than adding nothing like you did in the other cases?

Reply:
You need to return null to force the serializer to return the result of the visitor instead of the navigator: https://github.com/schmittjoh/serializer/blob/1.13.0/src/JMS/Serializer/Serializer.php#L183

`// noop` would have worked here as well but I wanted to be explicit about it.

Reply:
Ok, good reason to have it